### PR TITLE
Changed logic for loading journal files to include all logs for the last week

### PR DESCRIPTION
**Pull request by jberanek**

### CODE REVIEW

File: controller.py [hunk 1]
```diff
@@ -38,11 +41,23 @@ def rebuild_from_logs(self):
...
+        # Calculate a week ago, as a regex
+        now = datetime.datetime.now()
+        week_ago = now - timedelta(days=7)
+        week_ago_regex = week_ago.strftime('\.%y%m%d')
+
+        # Then build an array with the wanted logs
+        wanted_logs = []
+        found_first_log = False
         for log in logs:
+            logger.info(f"Checking {log} with {week_ago_regex}")
```
Comment (jberanek) [inline controller.py]: Unintentionally left this bit of debug in, probably too verbose!

**Review by jberanek — COMMENTED** [review]
You know, my logic is flawed...let me think on this a little longer.

I need to take the log filename and turn it into a date instead!

**Comment by jberanek** [comment]
Improved the logic now

File: controller.py [hunk 2]
```diff
@@ -38,11 +41,30 @@ def rebuild_from_logs(self):
...
+              if match:
+                  log_date = datetime.datetime(int(match.group(1))+2000, # Yucky 2 digit dates!
+                                               int(match.group(2)),
+                                               int(match.group(3)),
+                                               int(match.group(4)))
+                  if log_date >= week_ago:
+                      found_first_log = True
+
+            if found_first_log:
+                wanted_logs.append(log)
```
Comment (lunarplasma) [inline controller.py]: If you could stick this whole bit of code into a separate function, I'd say it looks pretty damned good.

File: controller.py [hunk 3]
```diff
@@ -38,11 +41,30 @@ def rebuild_from_logs(self):
...
+        # Calculate a week ago
+        now = datetime.datetime.now()
+        week_ago = now - timedelta(days=7)
+
+        # Then build an array with the wanted logs
+        wanted_logs = []
+        found_first_log = False
         for log in logs:
+            if not found_first_log:
+              match = re.search(r"^Journal\.(\d\d)(\d\d)(\d\d)(\d\d)", log)
```
Comment (lunarplasma) [inline controller.py]: I'd compile this outside the `for` loop, then use the compiled matcher object inside of it. 

Something like this, I think:

```py
matcher = re.compile(r"^Journal\.(\d\d)(\d\d)(\d\d)(\d\d)")
for log in logs:
  match = matcher.search(log)
```

**Review by lunarplasma — COMMENTED** [review]
Just a couple of comments here. Thanks for quashing yet another bug!

**Pull request merged**